spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/RowLevelCommandScanRelationPushDown.scala

```diff
@@ -68,6 +68,20 @@ object RowLevelCommandScanRelationPushDown extends Rule[LogicalPlan] with Predic
       command.withNewRewritePlan(newRewritePlan)
   }
 
+  private def pushFilters(
+      cond: Expression,
+      scanBuilder: ScanBuilder,
+      tableAttrs: Seq[AttributeReference]): (Seq[Filter], Seq[Expression]) = {
+
+    val tableAttrSet = AttributeSet(tableAttrs)
+    val filters = splitConjunctivePredicates(cond).filter(_.references.subsetOf(tableAttrSet))
```

Comment:
Was this what was preventing pushdown before? We weren't filtering out expressions that referenced columns outside of the table?

Reply:
Yes, we did not split the condition before into parts and did not remove filters that referenced both tables.

Reply:
@szehon-ho ˆˆ

This comment provides a little bit more info to answer your question above.
We treated `t.id = s.id and t.dep IN ('hr')` as a single predicate that couldn't be converted as it referenced both tables. Instead, we now split it into parts and convert whatever we can (i.e. `t.dep IN ('hr')` in this case).